modules/nf-core/kraken2/build/main.nf

```diff
@@ -0,0 +1,46 @@
+process KRAKEN2_BUILD {
+    tag "$meta.id"
+    label 'process_medium'
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-f8c4015c836dd3ce5c118cfed97ec8259bab9e9d:2e0b144854b4a3d69b5df7a0340a60db846cc8bf-0':
+        'biocontainers/mulled-v2-f8c4015c836dd3ce5c118cfed97ec8259bab9e9d:2e0b144854b4a3d69b5df7a0340a60db846cc8bf-0' }"
+
+    input:
+    tuple val(meta), path(db)
+
+    output:
+    tuple val(meta), path("$prefix"), emit: db
+    path "versions.yml"             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    kraken2-build \\
+        --build \\
+        $args \\
+        --threads $task.cpus \\
+        --db ${db}
```

Comment:
Is this expected to match `prefix`? I guess you might need to stage in path(db) as whatever you want into a directory called `${prefix}`... to make sure it gets emitted correctly?

Also might be worht having an optional clean up command here, given the large amount of space it takes up (use a value channel to get a boolean 'switch' to activate it)

Reply:
1. I did not understand this part. The files that should be generated are  actually there (in the database directory). Also` ${prefix}` is defined by `${meta.id}`, as long as this is unchanged the output directory should be fine.
2.

Reply:
1. What I mean is that the final database directory is `--db ${db}` (which is also the input). You do not use `${prefix}` anywhere in the command itself. However what you are emitting from the is `${prefix}`.

If someone in their config file modifies the prefix, e.g. `ext.prefix = "blah"`, Nextflow will try to emit `blah` which will not exist, as the directory is still called `db`.  Therefore we cannot guarantee that prefix == `{meta.id}`.

Does that make sense? So we need to find a way either emit the input, or rename the output with prefix - I guess `mv ${db} ${prefix}`would be simple enough, except if they _don't_ change it it will complain?

2. Perfect :+1:

Reply:
> 1. What I mean is that the final database directory is `--db ${db}` (which is also the input). You do not use `${prefix}` anywhere in the command itself. However what you are emitting from the is `${prefix}`.
> 
> 
> If someone in their config file modifies the prefix, e.g. `ext.prefix = "blah"`, Nextflow will try to emit `blah` which will not exist, as the directory is still called `db`. Therefore we cannot guarantee that prefix == `{meta.id}`.
> 
> Does that make sense? So we need to find a way either emit the input, or rename the output with prefix - I guess `mv ${db} ${prefix}`would be simple enough, except if they _don't_ change it it will complain?
> 
>     2. Perfect 👍

This might work: 

    kraken2-build \\
        --build \\
        $args \\
        --threads ${task.cpus} \\
        --db ${db}
    $runclean
    if [[ \$(basename ${db}) != "${prefix}" ]]; then
        mv ${db}/* ${prefix}
    fi